roadmap/implementers-guide/src/node/validity/assignments.md

```diff
@@ -0,0 +1,101 @@
+
+There is no sampling process for `RelayVRFDelay` and `RelayEquivocation`.  We instead run them on specific candidates and they compute a delay from their VRF output.  `RelayVRFDelay` inputs a `RelayVRFStory` runs for all candidates entering availability, equivalently all candidates leaving some availability core.   `RelayEquivocation` run only candidate lock equivocations, and inputs their block hashes via the `RelayEquivocation` story.
+
+`RelayVRFDelay` and `RelayEquivocation` both compute their output with `schnorrkel::vrf::VRFInOut::make_bytes` using the context "tranche" and reduce the result modulo `num_delay_tranches + zeroth_delay_tranche_width`, and consolidate results 0 through `zeroth_delay_tranche_width` to be 0.  In this way, they ensure the zeroth delay tranche has `zeroth_delay_tranche_width+1` times as many assignments as any other tranche.
+
+As future work (or TODO?), we should merge assignment notices with the same delay and story using `vrf_merge`.  We cannot merge those with the same delay and different stories because `RelayEquivocationStory`s could change but `RelayVRFStory` never changes. 
+
+## Announcer and Watcher/Tracker
+
+We track all validators' announced approval assignments for each candidate associated to each relay chain block.  
+
+We permit at most one assignment per candidate per story per validator, so one validator could be assigned under both the `RelayVRFDelay` and `RelayEquivocation` criteria, but not under both `RelayVRFModulo` and `RelayVRFDelay` criteria, since those both use the same story.  We permit only one approval vote per candidate per validator, which counts for any applicable criteria. 
+
+We announce, and start checking for, our own assignments when their delay is reached, but only if the tracker says the assignee candidate requires more approval checkers.  We never announce an assignment we believe unnecessary because early announcements gives an adversary information.  All delay tranche zero assignments always get announced, which includes all `RelayVRFModulo` assignments.
+
+We have a "no show" timeout longer than one relay chain slot, so at least six seconds, during which we expect approval checks should succeed in reconstructing the candidate block, in redoing its erasure coding to check the candidate receipt, and in rechecking the candidate block itself.  
+
+We consider a validator a "no show" if they do not approve or dispute within this "no show" timeout from our receiving their assignment notice.  We worry "no shows" represent a validator under denial of service attack, presumably to prevent it from reconstructing the candidate, but perhaps delaying it form gossiping a dispute too.  We always replace "no shows" by adding one entire extra delay tranche worth of validators, so such attacks always result in additional checkers.  
```

Comment:
What does replacement mean in this context? This happens off-chain, right. What is the significance of any single validator adding an extra tranche of validators? Does that make its way on-chain somehow?

Reply:
I'd move any more details discussion about the asymptotics to https://github.com/w3f/research-internal/issues/515 

We need the same logic on-chain as off-chain to know if a candidate is approved on-chain, but.. 

I'd expect no shows would usually resolve themselves fast, so the on-chain logic could abandon the replacements too.  We'll discuss incentives for replacements later, but obviously they impact our abandoning replacements.